staging/src/k8s.io/apiserver/pkg/admission/interfaces.go

```diff
@@ -49,6 +49,8 @@ type Attributes interface {
 	GetKind() schema.GroupVersionKind
 	// GetUserInfo is information about the requesting user
 	GetUserInfo() user.Info
+	// GetAnnotations returns an annotation map for the admission chain, both ValidationInterface and MutationInterface are allowed to insert new Annotations.
+	GetAnnotations() map[string]string
```

Comment:
@liggitt @sttts 
Please take a look.

My question:
Why not `SetAnnotations(key, value string)`?
If we use SetAnnotations:
1. annotations set by one admission controller will not exposed to each other. (What if an admission needs to read annotations from others?)
2. we can forbid one admission controller to modify annotations set by others.

GetAnnotations() is consistent with GetObject().

Reply:
> (What if an admission needs to read annotations from others?)

It shouldn't.

> we can forbid one admission controller to modify annotations set by others.

We trust them anyway.

I am not a fan of a completely free form string map. We should at least define a convention for the keys or even enforce that, e.g. `admission.k8s.io/PodSecurityPolicy-<key>: <value>` and `admission.k8s.io/MutatingWebhook-your.webhook.company.com-<key>: <value>`

Reply:
I add SetAnnotations() to enforce the annotion format.

Reply:
We can remove GetAnnotations(). If so, users can only set Annotation with function SetAnnotations(). And we need to change the signature of fucntion:
https://github.com/kubernetes/kubernetes/pull/58143/commits/3e253dc5f5f432c6f111d47fb299a7281cd440fd#diff-ba2d46d03943b63d3c3add85abff408bR157

Reply:
I would like to avoid opening this channel as it makes reordering of plugins hard later. So better remove it.

Reply:
Done.